### ISSUE TITLE
Moving to 2.0 

### ISSUE THREAD

**Issue by stomita** [jsforce/jsforce#889]
All,

I am on the way to update this library code in ver 2.0, which introduces TypeScript-based static typing in JSforce source code.

https://github.com/jsforce/jsforce/tree/2.0
https://github.com/jsforce/jsforce/tree/typescript-init

This is the most meaningful update from the release of 1.0, so I expect an experienced developer should lead it first. The update is still in progress, but not sure when to release as my own company business becomes so busy recently.

In parallel, I will update for the critical issues on 1.0-based branches, but I'm not willing to add any new features on 1.x as it makes much difficult to move to 2.0. The remaining PRs, I haven't seen all of them yet, would be kept as so and wait after 2.0 releases, except the ones which involves vulnerabilities or serious availabilities.

I know all of you are expecting much active commitment on this lib, but it is not always possible as we all have our own dedicated businesses. Additionally, as the lib is in drastic change, merely adding committers/maintainers might not be the solution in this case, as it needs another overhead to communicate within these members. However, the lib is not dead, and I'm now struggling to bring more sophisticated DX to the lib users.

**Comment by paustint** [comment]
@stomita - Thank you so much for the update and all of the work you have put in to this extremely important and amazing library.

> which introduces TypeScript-based static typing 

This is amazing! I really look forward seeing the progress.

**Comment by GarethSharpe** [comment]
Thanks for your work. If you are looking for external contributors, please reach out. Our company may be interested in contributing back to this library if new features need to be build out. 

Looking forward to the changes!

**Comment by stomita** [comment]
2.0.0-alpha.1 is published to npm. You can install it by `npm i jsforce@2.0.0-alpha.1`.
This is alpha version so not all features in 1.x are included.

Also [blogged](https://jsforce.github.io/blog/posts/20191216-jsforce20-alpha-preview-with-schema-type-feature.html) about the Schema Type feature in 2.0.

**Comment by paustint** [comment]
@stomita - This is awesome news, thank you so much for all the work you do for the community!

**Comment by yayajacky** [comment]
@stomita You are amazing!

**Comment by amphro** [comment]
I created a new issue to track the status of V2. #1137
